### DISCUSSION TITLE
Mi Band 5 syncing very slow with Watchdrip and xDrip

### DISCUSSION THREAD

**Discussion by kinir297** [NightscoutFoundation/xDrip#3074] [Bug reports] [answered]
Hi everyone! I got this issue with my Mi Band 5 syncing very slow (it says syncing for about 1 minute every time) before updating my blood sugar, sometimes the syncing even fails after trying to sync for 1 min. When I force stop Watchdrip+ and open it again, the syncing is quicker (about 5 seconds) but after some time passes (about an hour but sometimes less or more) the same issue happens again. I also tried reinstalling Watchdrip+ and factory resettig the Mi Band 5, that temporarily fixed the issue but in a few days it started again. xDdrip is okay and shows results quickly, so I think the issue is in Watchdrip+ or my device, I'm using Samsung A53 5G but the phone works perfectly with xDrip so I believe that's not the issue. I have Dexcom One with transmitter, they work okay too.
Does someone have experience with this and any ideas how to fix it?
Thanks in advance!
![20230909_230300](https://github.com/NightscoutFoundation/xDrip/assets/144566628/8e8bf1a8-4d88-48ee-9e0a-43cdc0226edb)
![Screenshot_20230910_014434_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/144566628/b1fc8469-de0c-4890-abc8-5e62685d277b)

**Comment by Navid200** [comment]
You should ask Watchdrip.  
If the cause of the problem is xDrip, we can make changes.  But, your process should start at the Watchdrip end, not the other way around.

**Reply by kinir297** [reply]
I see, I thought the same people are working on both xDrip and Watchdrip, that's why I wrote here. In this case should I contact Watchdrip the same way in their Github? I'm new here and not really sure how it works sorry.

And in general thank you for all your work and dedication, these apps really changed my life and made it much easier now, even with this issue they are of great help!

**Reply by Navid200** [reply, marked as answer]
I believe you can open an issue here to communicate with the Watchdrip developer:  https://github.com/bigdigital/watchdrip/issues

**Comment by kresimirjedud** [comment]
Hi Navid,

One question regarding watch integration. I'm having problem with a new
watch and watchdrip which worked just fine with old watch and an idea just
came on how to solve the watch 'problem' globally. Everyone is fighting the
same battle: which watch works with xdrip ...? I noticed that nitifications
like ' Low predicts in xyz minutes' and notifications like 'Missed readings
...' can be displayed in (probably) any smartwatch notifications (not as a
watchface but as a notification, but this is good enough). Perhaps xdrip
could be configured to send readings in the same way to the phone and then
the phone could send them as notifications to smartwatch. Perhaps this is
not too much work to do since there is already engine doing this for above
mentioned. Thank you for your reply and time!

Regards

ned, 10. ruj 2023. 01:36 Navid ***@***.***> je napisao:

**Reply by Navid200** [reply]
WatchDrip used to be a part of xDrip.  But, it eventually became a companion app.  Now. new watches can come to the market and they can be covered still with no need to modify xDrip.
This is a good thing for everyone.

xDrip is an app that contains a lot of functionalities.  Maintaining all of those takes a lot of time.  It is not practical to expect one person, the owner of the repository, to oversee all of that work.  
Any feature request for miband should go to the existing companion app.

My hope is that the entire wearOS will also move to another companion app.  
Adding more watch-related features to xDrip is a move in the wrong direction in my opinion.

All PRs are considered.  If you want to do what you are suggesting, go ahead and open a PR.
If you are asking us to do that, please note that I have multiple xDrip tasks that I have put on hold because there is no  time to execute them all.  I have no bandwidth for anything more.  Everyone else working on xDrip is in the same boat.  We are all overloaded with tasks.
I assure you we don't have a shortage of ideas.
